### PR TITLE
feat(clickhouse): column constraints

### PR DESCRIPTION
**Pull request by pkit**
add ClickHouse-specific column constraints for `CREATE TABLE`

- MATERIALIZED
- EPHEMERAL

see: https://clickhouse.com/docs/en/sql-reference/statements/create/table#default_values

### CODE REVIEW

**Comment by georgesittas** [comment]
I think `ALIAS` was recently added by @VaggelisD, see https://github.com/tobymao/sqlglot/commit/f88640b8df22e29ad2fa845b580cf78ad4fb2262

File: sqlglot/tokens.py [hunk 1]
```diff
@@ -293,6 +294,7 @@ class TokenType(AutoName):
     MAP = auto()
     MATCH_CONDITION = auto()
     MATCH_RECOGNIZE = auto()
+    MATERIALIZED = auto()
```
Comment (georgesittas) [inline sqlglot/tokens.py:297]: These new tokens aren't used anywhere, let's get rid of them

**Comment by pkit** [comment]
Yes, `ALIAS` is a ComputeColumn. Interesting how I missed that. Will remove my implementation.

**Comment by pkit** [comment]
Done, now `MATERIALIZED` is just a `persisted` flavor of a `ComputedColumn`

File: sqlglot/parser.py [hunk 2]
```diff
@@ -4513,6 +4517,15 @@ def _parse_column_def(self, this: t.Optional[exp.Expression]) -> t.Optional[exp.
...
             )
+        elif self._match_text_seq("MATERIALIZED"):
+            constraints.append(
+                self.expression(
+                    exp.ComputedColumnConstraint,
+                    this=self._parse_conjunction(),
+                    persisted=True,
+                    not_null=True,
+                )
+            )
```
[the remark this answers is not in the document]
Comment (pkit) [reply sqlglot/parser.py:4528]: Yep, I though how to consolidate, this one looks nice! I forgot we can do `self._prev.text`

**Pull request merged**